NeoMathEngine/src/CPU/x86/CpuX86MathEngineBlasMkl.cpp

```diff
@@ -88,18 +80,14 @@ void CCpuMathEngine::multiplyMatrixByMatrixAndAdd( const CConstFloatHandle& firs
 #endif
 }
 
-void CCpuMathEngine::multiplyMatrixByTransposedMatrix(const CConstFloatHandle& firstHandle, int firstHeight,
-	int firstWidth, int firstRowSize, const CConstFloatHandle& secondHandle, int secondHeight, int secondRowSize,
-	const CFloatHandle& resultHandle, int resultRowSize, int resultBufferSize)
+void CCpuMathEngine::multiplyMatrixByTransposedMatrix(const float* first, int firstHeight,
+	int firstWidth, int firstRowSize, const float* second, int secondHeight, int secondRowSize,
+	float* result, int resultRowSize, int resultBufferSize)
 {
 	ASSERT_EXPR(firstWidth <= firstRowSize);
```

Comment:
For the sake of optimization all of the `ASSERT_EXPR` can be removed from **private** versions.

*or, replaced with `assert` from STL which is removed from non-debug configurations